### ISSUE TITLE
CIFAR10, CIFAR100 reproduce results

### ISSUE THREAD

**Issue by insikk** [KaiyangZhou/deep-person-reid#230]
With some modification of codes, I was able to run this code base for CIFAR10 classification task. I followed ResNet paper for hyperparameters. Even though some values are different, I was expecting error rate less than 10%, but I got 12% error after 165 epochs of training. 

Can I get exact hyperparameters to reproduce CIFAR10 error rate result that was on Table 8 of its paper?

![image](https://user-images.githubusercontent.com/2110359/65232411-0e72b600-db0c-11e9-9ba4-8b8e8d23fdad.png)

```
model:
  name: 'osnet_x1_0'
  pretrained: False
  feature_dim: -1 # no fc layer. Direct classificatino from pooling. 

data:
  type: 'image'
  sources: ['cifar10']
  targets: ['cifar10']
  height: 32
  width: 32
  combineall: False
  transforms: ['random_flip', 'random_crop'] # 4px padding and random crop to become 32x32
  save_dir: 'log/cifar10_softmax_cosinelr_resnet_setting_0919'

sampler:
  train_sampler: RandomSampler

loss:
  name: 'softmax'
  softmax:
    label_smooth: False
    temperature: 1.0

sgd:
  nesterov: False

train:
  optim: 'sgd'  
  lr: 0.01 # after 1 epoch warmup, target lr become 0.1 as stated in the paper.
  max_epoch: 165
  warmup_epoch: 1
  multiplier_after_warmup: 10
  weight_decay: 0.0001
  batch_size: 128
  fixbase_epoch: 0
  open_layers: ['classifier']
  lr_scheduler: 'multi_step'
  stepsize: [82, 123]
  gamma: 0.1
  base_lr_mult: 1.0

test:
  batch_size: 300
  dist_metric: 'euclidean'
  normalize_feature: False
  evaluate: False
  eval_freq: 1 # evaluate after every 1 epoch.
  ranks: [1, 3, 5, 9] # cmc ranks
  rerank: False
  visactmap: False
  eval_categorisation: True
```

**Comment by KaiyangZhou** [comment]
follow this code https://github.com/kuangliu/pytorch-cifar/blob/master/models/resnet.py to modify osnet for cifar (e.g. the first layer is 3x3)

other training params follow exactly the same as the cited paper

**Comment by insikk** [comment]
Thank you for your comment. I followed your suggestion (changing first convolution layer parameter, and removing max pooling after the first conv), then I got the result similar to what reported in the OSNet paper.



Here are my training curve. 

![image](https://user-images.githubusercontent.com/2110359/65400631-46287900-ddfe-11e9-9ddc-8956584dc1a2.png)

**Comment by szxczyc** [comment]
can u share the code about hoo to add warmup to `lr_scheduler.py`?thank u
